lib/_stream_writable.js

```diff
@@ -155,11 +155,12 @@ Writable.prototype.pipe = function() {
 };
 
 
-function writeAfterEnd(stream, cb) {
-  var er = new Error('write after end');
-  // TODO: defer error events consistently everywhere, not just the cb
-  stream.emit('error', er);
-  process.nextTick(cb, er);
+function writeAfterEndErr(stream, cb) {
+  const err = new Error('write after end');
+  process.nextTick(() => {
+    stream.emit('error', err);
+    cb(err);
+  });
```

Comment:
we do not support arrow functions at the moment, and it would be tricky to do so in a performant manner, that being said, you could change this code to be, 

```js

function writeAfterEnd(stream, cb) {
  const err = new Error('write after end');
  stream.emit('error', err);
  cb(err);
}

and then replace the use of writeAfterEndErr(this, cb) with 

```js
 process.nextTick(writeAfterEnd, this, cb);
```

and this would avoid creating an unnecessary closure